test/e2e/webdriver/driver.js

```diff
@@ -546,6 +546,43 @@ class Driver {
     throw new Error(`No window with title: ${title}`);
   }
 
+  async switchToWindowWithUrl(
+    url,
+    initialWindowHandles,
+    delayStep = 1000,
+    timeout = this.timeout,
+    { retries = 8, retryDelay = 2500 } = {},
+  ) {
+    let windowHandles =
+      initialWindowHandles || (await this.driver.getAllWindowHandles());
+    let timeElapsed = 0;
+
+    while (timeElapsed <= timeout) {
+      for (const handle of windowHandles) {
+        const handleUrl = await retry(
+          {
+            retries,
+            delay: retryDelay,
+          },
+          async () => {
+            await this.driver.switchTo().window(handle);
+            return await this.driver.getCurrentUrl();
+          },
+        );
+
+        if (handleUrl === `${url}/`) {
+          return handle;
+        }
+      }
+      await this.delay(delayStep);
+      timeElapsed += delayStep;
+      // refresh the window handles
+      windowHandles = await this.driver.getAllWindowHandles();
```

Comment:
is it possible to do this all within the `await retry`?

```
await retry(
          {
            retries,
            delay: retryDelay,
          },
          async () => {
            windowHandles = await this.driver.getAllWindowHandles();
            for (const handle of windowHandles) {
              await this.driver.switchTo().window(handle);
              const handleUrl await this.driver.getCurrentUrl();
              if (handleUrl === `${url}/`) {
                return true;
              }
            }
          },
        );

```

Reply:
something like that?

Reply:
cc @tmashuang ☝️

Reply:
I tried what @jiexi proposed and It didn't seem to work. Looks like it wasn't able to break out of the `retry()`

Reply:
ahhh, i'm sorry. Thank you for taking on the burden of trying it. Apologies 🙏